pkg/cmd/doctor/run.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package doctor
+
+import (
+	"compress/flate"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/mholt/archiver"
+	"github.com/okteto/okteto/pkg/config"
+	"github.com/okteto/okteto/pkg/k8s/pods"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	"k8s.io/client-go/kubernetes"
+)
+
+//Run runs the "okteto status" sequence
+func Run(ctx context.Context, dev *model.Dev, c *kubernetes.Clientset) error {
+	z := archiver.Zip{
+		CompressionLevel:       flate.DefaultCompression,
+		MkdirAll:               true,
+		SelectiveCompression:   true,
+		ContinueOnError:        true,
+		OverwriteExisting:      true,
+		ImplicitTopLevelFolder: true,
+	}
+
+	summaryFilename, err := generateSummaryFile()
+	if err != nil {
+		return err
+	}
+	defer os.Remove(summaryFilename)
+
+	remoteLogsPath, err := generateRemoteSyncthingLogsFile(ctx, dev, c)
+	if err != nil {
+		log.Yellow("Failed to query remote syncthing logs: %s", err)
+	}
+	if remoteLogsPath != "" {
+		defer os.Remove(remoteLogsPath)
+	}
+
+	now := time.Now()
+	archiveName := fmt.Sprintf("okteto-doctor-%s.zip", now.Format("20060102150405"))
+	files := []string{summaryFilename, filepath.Join(config.GetHome(), "okteto.log"), config.GetSyncthingLogFile(dev.Namespace, dev.Name)}
+	if remoteLogsPath != "" {
+		files = append(files, remoteLogsPath)
+	}
+	if err := z.Archive(files, archiveName); err != nil {
+		log.Infof("error while archiving: %s", err)
+		return fmt.Errorf("couldn't create archive, please try again")
+	}
+
+	log.Information("Your doctor file was generated at %s", archiveName)
```

Comment:
don't call `log.Information` outside of the `cmd` package, so we can control UI elements there.  I think you can return the file and do the notification there.